util/plugins/eu.esdihumboldt.util/src/eu/esdihumboldt/util/geometry/interpolation/Interpolation.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2016 wetransform GmbH
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.util.geometry.interpolation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+
+/**
+ * Base class of interpolation
+ * 
+ * @author Arun
+ * @param <T> the type of interpolated Geometry
+ */
+public abstract class Interpolation<T extends Geometry> {
+
+	/**
+	 * Grid step size factor
+	 */
+	protected final int GRID_FACTOR = 2;
+	/**
+	 * Next coordinate distance factor.
+	 */
+	protected final double NEXT_COORDINATE_DISTANCE_FACTOR = 4;
+	/**
+	 * Distance of next coordinate from current coordinate
+	 */
+	protected final double NEXT_COORDINATE_DISTANCE;
+
+	/**
+	 * Maximum positional error
+	 */
+	protected final double MAX_POSITIONAL_ERROR;
+
+	/**
+	 * Coordinates of Geometry, which wanted to be interpolated
+	 */
+	protected final Coordinate[] rawGeometryCoordinates;
+
+	/**
+	 * Constructor
+	 * 
+	 * @param coordinates Coordinates of geometry that need to be interpolated
+	 * @param maxPositionalError maximum positional error for interpolation
+	 */
+	public Interpolation(Coordinate[] coordinates, double maxPositionalError) {
+		this.MAX_POSITIONAL_ERROR = maxPositionalError;
+		this.rawGeometryCoordinates = coordinates;
+		this.NEXT_COORDINATE_DISTANCE = NEXT_COORDINATE_DISTANCE_FACTOR * maxPositionalError;
+	}
+
+	/**
+	 * interpolate raw geometry
+	 * 
+	 * @return interpolated Geometry of type T or <code>null</code>
+	 */
+	public T interpolateRawGeometry() {
+		if (!validateRawCoordinates()) {
+			return null;
+		}
+
+		return interpolatedGeometry();
+	}
+
+	/**
+	 * validate raw geometry
+	 * 
+	 * @return true if validation successful else false.
+	 */
+	protected abstract boolean validateRawCoordinates();
+
+	/**
+	 * get interpolate raw geometry in to
+	 * 
+	 * @return interpolated Geometry
+	 * 
+	 */
+	protected abstract T interpolatedGeometry();
+
+	/**
+	 * relocate geometry coordinate to nearest grid point.
+	 * 
+	 * @param coordinate geometry coordinate
+	 * @return relocated grid coordinate
+	 */
+	protected Coordinate pointToGrid(Coordinate coordinate) {
```

Comment:
Because it is hard to create test cases to test the functionality of interpolation exactly, I think it would be good to test parts of the algorithm that can be easily unit tested. This method is a candidate for that, it is quite easy to come up with tests here and cover all the corner cases.